spec/unit/gateway/contentful_gateway_spec.rb

```diff
@@ -0,0 +1,203 @@
+      contentful_gateway.get_case_study(slug: slug)
+
+      assert_requested :get, case_study_url
+    end
+
+    it 'can send a get request to Contentful with a space id' do
+      contentful_gateway.get_case_study(slug: slug)
+
+      assert_requested :get, initial_url
+    end
+
+    it 'can send a get request to Contentful with authorization' do
+      contentful_gateway.get_case_study(slug: slug)
+
+      assert_requested :get, initial_url, headers: { 'Authorization' => 'Bearer woof' }
+    end
+
+    it 'can return a case study with a name' do
+      case_study = contentful_gateway.get_case_study(slug: slug)
+
+      expect(case_study.name).to eq('Case Study - Meow Primary School')
+    end
+
+    it 'can return a case study with a slug' do
+      case_study = contentful_gateway.get_case_study(slug: slug)
+
+      expect(case_study.slug).to eq(slug)
+    end
+
+    it 'can return a case study with an image' do
+      case_study = contentful_gateway.get_case_study(slug: slug)
+
+      expect(case_study.hero_image).to eq('//images.ctfassets.net/dog/school-corridor.jpeg')
+    end
+
+    it 'can return a case study with content' do
+      case_study = contentful_gateway.get_case_study(slug: slug)
+
+      expect(case_study.content).to eq([
+        {
+          type: :image,
+          data: {
+            url: "//images.ctfassets.net/dog/school-corridor.jpeg",
+            width: :full_bleed
+          }
+        },
+        {
+          type: :heading,
+          data: {
+            text: "This is a test Woof heading",
+            level: :heading_one,
+            bold: true
+          }
+        },
+        {
+          type: :paragraph,
+          data: {
+            text: "This is a Woof paragraph woof woof"
+          }
+        }
+      ])
+    end
+  end
+
+  context '#get_case_study (example two)' do
+    let(:space_id) { 'wolf' }
+    let(:access_token) { 'awoo' }
+    let(:slug) { 'sss-primary-school-case-study' }
+    let(:contentful_gateway) do
+      described_class.new(space_id: space_id, access_token: access_token)
+    end
+    let(:initial_url) do
+      "https://cdn.contentful.com/spaces/#{space_id}/environments/master/content_types?limit=1000"
+    end
+    let(:case_study_url) do
+      "https://cdn.contentful.com/spaces/#{space_id}/environments/master/entries?content_type=caseStudy&include=10&fields.slug=#{slug}"
+    end
+
+    before do
+      headers['Authorization'] = "Bearer #{access_token}"
+
+      stub_request(:get, initial_url)
+        .with(headers: headers)
+        .to_return(status: 200, body: response_with_no_items, headers: {})
+
+      response_with_items = File.open("#{fixtures_path}case_study_example_two.json", &:read)
+      stub_request(:get, case_study_url)
+        .with(headers: headers)
+        .to_return(status: 200, body: response_with_items, headers: {})
+    end
+
+    it 'can send a get request to Contentful' do
+      contentful_gateway.get_case_study(slug: slug)
```

Comment:
You could move this to a before block 

```
let(:case_study) { contentful_gateway.get_case_study(slug: slug) } 

before { case_study }

```

Reply:
We refactored this in https://github.com/DFE-Digital/single-place-for-education-frontend/pull/6/commits/e98cdfdface6a6d1dbfaf80805641e013f0ca08b. Is this alright?